src/catch2/internal/catch_dll_public.hpp

```diff
@@ -0,0 +1,40 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE_1_0.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+// SPDX-License-Identifier: BSL-1.0
+#ifndef CATCH_DLL_PUBLIC_HPP_INCLUDED
+#define CATCH_DLL_PUBLIC_HPP_INCLUDED
+
+#if defined _WIN32 || defined __CYGWIN__
+#    ifdef BUILDING_DLL
+#        ifdef __GNUC__
+#            define CATCH_DLL_PUBLIC __attribute__( ( dllexport ) )
+#        else
+#            define CATCH_DLL_PUBLIC \
+                __declspec( dllexport ) // Note: actually gcc seems to also
+                                        // supports this syntax.
+#        endif
+#    else
```

Comment:
So, I don't plan to merge this (sorry), but FYI for the next time you have to deal with this: the logic for Windows has to be tri-state.

* DLL build and export
* DLL build and import
* static build

because for more complex static linking setups you want neither import nor export annotations.

Reply:
Thank you so much for letting me know about this! It's useful information for the future!

Reply:
This file can also be generated by CMake's [`GenerateExportHeader`](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html) module.

See an example here https://github.com/friendlyanon/cxx-static-shared-example#export-macros